yt_dlp/extractor/yahoo.py

```diff
@@ -459,22 +458,24 @@ def _real_extract(self, url):
 class YahooJapanNewsIE(InfoExtractor):
     IE_NAME = 'yahoo:japannews'
     IE_DESC = 'Yahoo! Japan News'
-    _VALID_URL = r'https?://(?P<host>(?:news|headlines)\.yahoo\.co\.jp)[^\d]*(?P<id>\d[\d-]*\d)?'
+    _VALID_URL = r'https?://(?P<host>(?:news|headlines)\.yahoo\.co\.jp)(?:/articles/|[^\d]*)(?P<id>(?:[\da-f]{40}|\d[\d-]*\d))?'
```

Comment:
This is because the id can begin with a letter so `[^\d]*` would capture part of the of the id. In the case the id has letters it will come directly after `/articles/`